packages/frontend/src/pages/Users/components/UsersTable.tsx

```diff
@@ -1,121 +1,159 @@
+import UsersTablePagination from './UsersTablePagination';
+
+const roleOptions = [
+  { name: 'All users', value: UserRole.USER },
+  { name: 'Admins', value: UserRole.ADMIN },
+  { name: 'Forwarders', value: UserRole.FORWARDER },
+  { name: 'Quantifiers', value: UserRole.QUANTIFIER },
+];
 
 const UsersTable = (): JSX.Element => {
+  const allAdminUsers = useRecoilValue(AllAdminUsers);
+  const allForwarderUsers = useRecoilValue(AllForwarderUsers);
   const allQuantifierUsers = useRecoilValue(AllQuantifierUsers);
-  const { removeRole } = useAdminUsers();
-
-  const deleteDialogRef = React.useRef(null);
-
-  const [isOpen, setIsOpen] = React.useState(false);
-  const [selectedQuantifier, setSelectedQuantifier] = React.useState<UserDto>();
-
-  const columns = React.useMemo(
-    () => [
-      {
-        Header: 'Id',
-        accessor: '_id',
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        Cell: (data: any): JSX.Element => {
-          return (
-            <div className="flex items-center w-full">
-              <div className="flex items-center">
-                <UserAvatar user={data.row.original} />
-              </div>
-              <div className="flex-grow p-3 whitespace-nowrap">
-                {data.row.original.accounts?.length > 0 ? (
-                  <div>{getUsername(data.row.original)}</div>
-                ) : (
-                  ''
-                )}
-                {shortenEthAddress(data.row.original.ethereumAddress)}
-              </div>
-            </div>
-          );
-        },
-      },
-    ],
-    []
+  const allUsers = useRecoilValue(AllUsers);
+  const [tableData, setTableData] = React.useState<UserDto[]>();
+  const [selectedRole, setSelectedRole] = React.useState<UserRole>(
+    UserRole.USER
   );
+  const [filter, setFilter] = React.useState<string>('');
+  const [filterType, setFilterType] = React.useState<string>('user');
+  const [page, setPage] = React.useState<number>(1);
+  const [lastPage, setLastPage] = React.useState<number>(0);
 
-  const options = {
-    columns,
-    data: allQuantifierUsers ? allQuantifierUsers : [],
-  } as TableOptions<{}>;
-  const tableInstance = useTable(options);
+  React.useEffect(() => {
+    if (allUsers) {
+      setTableData(allUsers);
+    }
+  }, [allUsers]);
 
-  const { getTableProps, getTableBodyProps, rows, prepareRow } = tableInstance;
+  React.useEffect(() => {
+    if (filter && tableData) {
+      let filteredData: UserDto[];
+      if (filterType === 'user') {
+        filteredData = tableData.filter((user) => {
+          const username = getUsername(user)?.toLowerCase();
+          if (username?.includes(filter.toLocaleLowerCase())) {
+            return user;
+          }
+        });
+      } else {
+        filteredData = tableData.filter((user) => {
+          if (
+            user.ethereumAddress?.toLocaleLowerCase() ===
+            filter.toLocaleLowerCase()
+          ) {
```

Comment:
The ethereumAddress search doesn't seem to be working. I can't get any results to appear.

Also, perhaps it should also check if the ethereumAddress *includes* the search text, instead of *equaling* the search text (just like you're doing with the user name search above).